components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	local teamTemplateDate = timestamp
+	-- If date is epoch, resolve using tournament dates instead
+	-- Epoch indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not 1970-01-01
+	if teamTemplateDate == DateExt.epochZero then
+		teamTemplateDate = Variables.varDefaultMulti(
+			'tournament_enddate',
+			'tournament_startdate',
+			TODAY
+		)
+	end
+
+	Opponent.resolve(opponent, teamTemplateDate)
+
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+
+	if record.type == Opponent.team then
+		record.match2players = CustomMatchGroupInput._readTeamPlayers(record, record.players)
+	end
+
+	return record
+end
+
+function CustomMatchGroupInput._readTeamPlayers(opponent, playerData)
+	local players = CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+	opponent.players = nil
+
+	if Table.isEmpty(players) then
+		players = CustomMatchGroupInput._getPlayersFromVariables(opponent.name)
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+	local players = {}
+	playerData = Json.parseIfString(playerData) or {}
+
+	for prefix, displayName in Table.iter.pairsByPrefix(playerData, 'p') do
+		local name = mw.ext.TeamLiquidIntegration.resolve_redirect(Logic.emptyOr(
+			playerData[prefix .. 'link'],
+			displayName
+		))
+
+		table.insert(players, {
+			name = name,
+			displayname = displayName,
+			flag = Flags.CountryName(playerData[prefix .. 'flag']),
+		})
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getPlayersFromVariables(teamName)
+	local players = {}
+
+	local playerIndex = 1
+	local prefix = teamName .. '_p' .. playerIndex
+	local playerName = Variables.varDefault(prefix)
+	while String.isNotEmpty(playerName) do
+		table.insert(players, {
+			name = playerName,
+			displayname = Variables.varDefault(prefix .. 'dn'),
+			flag = Flags.CountryName(Variables.varDefault(prefix .. 'flag')),
+		})
+
+		playerIndex = playerIndex + 1
+		prefix = teamName .. '_p' .. playerIndex
+		playerName = Variables.varDefault(prefix)
+	end
+
+	return players
+end
+
+--[[
+
+MapInput functions
+
+]]--
+function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	local map = Json.parseIfString(match['map' .. mapIndex]) or {}
+
+	if Table.isEmpty(map) then
+		match['map' .. mapIndex] = nil
+		return match, subGroupIndex
+	end
+
+	-- CR has no map names, use generic one instead
+	map.map = 'Set ' .. mapIndex
+
+	-- set initial extradata for maps
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+	}
+
+	-- inherit stuff from match data
+	map.type = Logic.emptyOr(map.type, match.type)
+	map.liquipediatier = match.liquipediatier
+	map.liquipediatiertype = match.liquipediatiertype
+	map.game = Logic.emptyOr(map.game, match.game)
+	map.date = Logic.emptyOr(map.date, match.date)
```

Comment:
To commons?

Reply:
not opposed but do not have the time atm